### PR TITLE
Jsdoc-convert - final refactoring

### PR DESCRIPTION
**Pull request by Siemienik**
### Context
Blocked by: #8607 

Extract modules for parser and renderer. That makes main file (index.mjs) much more manageable. 
Extract two postProcessors to own files.
Move logic outside indexes, because they are module's API.
Update dependency-flow.png

Before | After
---|---
image | image

[skip changelog]

### How has this been tested?
Should generate the same result as before changes.

### Types of changes
- Refactoring
- 
### Related issue(s):
1. #8335 
2.
3.

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

File: docs/.vuepress/tools/jsdoc-convert/index.mjs [hunk 1]
```diff
@@ -1,129 +1,22 @@
...
-
+/* eslint-disable no-useless-escape */
 /// shared dependencies
 import utils from '../utils.js';
 
 /// internal dependencies
-import configuration from './configuration.mjs';
-import { isJsdocPlugin } from './predictors.mjs';
-import { buildHeaderWriter } from './seo.mjs';
+import configuration from './configuration.js';
```
Comment (budnix) [inline docs/.vuepress/tools/jsdoc-convert/index.mjs:6]:
```suggestion
import configuration from './configuration.mjs';
```
On this branch, I see two files with the same content:`configuration.js` and `configuration.mjs`. I guess that we can use `.mjs` format and remove the `.js` file?

File: docs/.vuepress/tools/jsdoc-convert/index.mjs [hunk 2]
```diff
@@ -1,129 +1,22 @@
-/// npm dependencies
-import fs from 'fs';
-
+/* eslint-disable no-useless-escape */
```
Comment (budnix) [inline docs/.vuepress/tools/jsdoc-convert/index.mjs:1]:
```suggestion
```

**Review by budnix — CHANGES_REQUESTED** [review]

File: docs/.vuepress/tools/jsdoc-convert/index.mjs (hunk 2 above)
Comment (Siemienik) [reply docs/.vuepress/tools/jsdoc-convert/index.mjs]: 3613e7f7c28eb7324c403791e50176897d7abbff

File: docs/.vuepress/tools/jsdoc-convert/index.mjs (hunk 1 above)
Comment (Siemienik) [reply docs/.vuepress/tools/jsdoc-convert/index.mjs]: 3613e7f7c28eb7324c403791e50176897d7abbff

**Review by budnix — APPROVED** [review]

**Pull request merged**